test/common/src/main/java/cloud/piranha/test/common/FreePortFinder.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2002-2022 Manorrock.com. All Rights Reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *   1. Redistributions of source code must retain the above copyright notice,
+ *      this list of conditions and the following disclaimer.
+ *   2. Redistributions in binary form must reproduce the above copyright
+ *      notice, this list of conditions and the following disclaimer in the
+ *      documentation and/or other materials provided with the distribution.
+ *   3. Neither the name of the copyright holder nor the names of its
+ *      contributors may be used to endorse or promote products derived from
+ *      this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package cloud.piranha.test.common;
+
+import java.io.IOException;
+import java.net.Socket;
+import java.util.Random;
+
+/**
+ *
+ * Finds a free port that isn't currently bound to by anything else.
+ */
+public class FreePortFinder {
```

Comment:
I didn't notice that. yes, I can use that dependency. My solution selects candidate ports randomly to avoid clashes if 2 threads are searching for a port at the same time and both would find the same free port while I don't see that the free-port-finder would do that (https://github.com/alexpanov/freeportfinder/blob/master/src/main/java/me/alexpanov/net/FreePortFinder.java) but it looks like it has some logic to prevent clashes. Let's see if it's enough.

Reply:
But I'll leave my free port finder logic in the Arquillian container, I don't want to bring another dependency to the Arquillian container just for that. Is that OK?

Reply:
Sounds good! 😉